src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs

```diff
@@ -38,6 +39,25 @@ public static IResourceBuilder<PostgresServerResource> AddPostgres(this IDistrib
         var passwordParameter = password?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-password");
 
         var postgresServer = new PostgresServerResource(name, userName?.Resource, passwordParameter);
+
+        string? connectionString = null;
+
+        builder.Eventing.Subscribe<ConnectionStringAvailableEvent>(postgresServer, async (@event, ct) =>
+        {
+            connectionString = await postgresServer.GetConnectionStringAsync(ct).ConfigureAwait(false);
+        });
+
+        builder.Services.AddHealthChecks().AddNpgSql(sp => connectionString!, name: $"{name}_check", configure: (connection) =>
+        {
+            // HACK: The Npgsql client defaults to using the username in the connection string if the database is not specified. Here
```

Comment:
Need to figure out if/what we can do better here.

Reply:
I feel like we're missing a primitive here. Should this instead be health checking the database instead of the server? Shouldn't we also add health checks for each database resource?

Reply:
I think we do need to do that.

Reply:
The initializer? Lets queue it up for after WaitFor is in. It intersects with it but it should be  a discrete mechanism like health check integration.

Reply:
No, if I wait for db then it should wait for the parent resource (the db server)'s health check to be 👍🏾

Reply:
OK so what I've done is I've left the health check in for the database. My thinking is by doing this we give folks flexibility. If folks truly want to wait for the database to be healthy they can do that with no effort.

I think what people will likely do is something like we have in the waitfor playground where they have some kind of worker that waits for the server to be ready and then initializes the database, and other services that depend on the database will WaitFor that.